### PR TITLE
Upgrade to sbt 1.3.6, release sbt-dotty 0.4.0

### PR DESCRIPTION
**Pull request by smarter**

### CODE REVIEW

File: staging/src/scala/quoted/staging/QuoteDriver.scala [hunk 1]
```diff
@@ -64,17 +67,29 @@ private class QuoteDriver(appClassloader: ClassLoader) extends Driver {
...
+    def loop(cl: ClassLoader, suffixClasspath: String): String =
+      cl match {
+        case cl: URLClassLoader =>
+          val updatedClasspath = cl.getURLs
+            .map(url => Paths.get(url.toURI).toAbsolutePath.toString)
+            .mkString(
+              "",
+              File.pathSeparator,
+              if (suffixClasspath.isEmpty) "" else File.pathSeparator + suffixClasspath
+            )
```
Comment (nicolasstucki) [inline staging/src/scala/quoted/staging/QuoteDriver.scala]:
```
  val updatedClasspath = cl.getURLs
            .map(url => Paths.get(url.toURI).toAbsolutePath.toString)
  val fullClasspath =
    if (suffixClasspath.isEmpty) updatedClasspath else updatedClasspath :+ suffixClasspath
  fullClasspath.mkString(File.pathSeparator)
```

File: staging/src/scala/quoted/staging/QuoteDriver.scala [hunk 2]
```diff
@@ -64,17 +67,29 @@ private class QuoteDriver(appClassloader: ClassLoader) extends Driver {
...
+          loop(cl.getParent, updatedClasspath)
```
Comment (nicolasstucki) [inline staging/src/scala/quoted/staging/QuoteDriver.scala]: This implementation is quadratic while summing Strings. I will replace it with the version from https://github.com/lampepfl/dotty/pull/7957/files

**Comment by nicolasstucki** [comment]
@smarter `sjsJUnitTests` is failing

File: staging/src/scala/quoted/staging/QuoteDriver.scala [hunk 3]
```diff
@@ -67,29 +67,25 @@ private class QuoteDriver(appClassloader: ClassLoader) extends Driver {
...
+    def collectClassLoaderPaths(cl: ClassLoader): Unit = cl match {
+      case cl: URLClassLoader =>
+        collectClassLoaderPaths(cl.getParent)
+        classpathBuff ++=
+          cl.getURLs.iterator.map(url => Paths.get(url.toURI).toAbsolutePath.toString)
+      case _ =>
+    }
+    collectClassLoaderPaths(cl)
+    val systemClasspath = System.getProperty("java.class.path")
+    if (!systemClasspath.isEmpty)
```
Comment (smarter) [inline staging/src/scala/quoted/staging/QuoteDriver.scala]: This is wrong, it's going to pollute the classpath with whatever is on java.class.path which might be totally unrelated to the code we're running.

**Comment by gzoller** [comment]
Hello!  The classpath discovery fix needs to also go in dotty.tools.dotc.consumetasty.ConsumeTasty.  The original code is the same there as in QuoteDriver.

**Comment by smarter** [comment]
`sjsJUnitTests/test` fails linking on the CI because it can't find dotty-library, but it's working fine locally so I have no idea what could be going wrong, any hint on how to debug this @sjrd ?
```scala
[info] Fast optimizing /tmp/2/tests/sjs-junit/../out/bootstrap/sjsJUnitTests/scala-0.22/sjsjunittests-test-fastopt.js
[error] Referring to non-existent class dotty.runtime.Arrays$
[error]   called from org.scalajs.testsuite.compiler.ArrayTest.setArrayIndexOutOfBounds()void
```

**Comment by smarter** [comment]
Locally at least, the classpath seems OK, wdyt ?
```scala
[dotty@upgrade/sbt-1.3.6]> show sjsJUnitTests/test:fullClasspath
[info] Checking out Scala.js source version 1.0.0-RC2
[info] * Attributed(/home/smarter/opt/dotty/tests/sjs-junit/../out/bootstrap/sjsJUnitTests/scala-0.22/test-classes)
[info] * Attributed(/home/smarter/opt/dotty/tests/sjs-junit/../out/bootstrap/sjsJUnitTests/scala-0.22/classes)
[info] * Attributed(/home/smarter/opt/dotty/library-js/../out/bootstrap/dotty-library-bootstrappedJS/scala-0.22/classes)
[info] * Attributed(/home/smarter/.cache/coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/scala-library/2.13.1/scala-library-2.13.1.jar)
[info] * Attributed(/home/smarter/.cache/coursier/v1/https/repo1.maven.org/maven2/org/scala-js/scalajs-test-interface_2.13/1.0.0-RC2/scalajs-test-interface_2.13-1.0.0-RC2.jar)
[info] * Attributed(/home/smarter/.cache/coursier/v1/https/repo1.maven.org/maven2/org/scala-js/scalajs-junit-test-runtime_2.13/1.0.0-RC2/scalajs-junit-test-runtime_2.13-1.0.0-RC2.jar)
[info] * Attributed(/home/smarter/.cache/coursier/v1/https/repo1.maven.org/maven2/org/scala-js/scalajs-test-bridge_2.13/1.0.0-RC2/scalajs-test-bridge_2.13-1.0.0-RC2.jar)
[info] * Attributed(/home/smarter/.cache/coursier/v1/https/repo1.maven.org/maven2/org/scala-js/scalajs-library_2.13/1.0.0-RC2/scalajs-library_2.13-1.0.0-RC2.jar)
[success] Total time: 1 s, completed 14 janv. 2020 23:57:43
[dotty@upgrade/sbt-1.3.6]> show sjsJUnitTests/test:scalacOptions
[info] * -feature
[info] * -unchecked
[info] * -encoding
[info] * UTF8
[info] * -language:existentials,higherKinds,implicitConversions
[info] * -bootclasspath
[info] * /usr/lib/jvm/java-8-openjdk-amd64/jre/lib/resources.jar:/usr/lib/jvm/java-8-openjdk-amd64/jre/lib/rt.jar:/usr/lib/jvm/java-8-openjdk-amd64/jre/lib/sunrsasign.jar:/usr/lib/jvm/java-8-openjdk-amd64/jre/lib/jsse.jar:/usr/lib/jvm/java-8-openjdk-amd64/jre/lib/jce.jar:/usr/lib/jvm/java-8-openjdk-amd64/jre/lib/charsets.jar:/usr/lib/jvm/java-8-openjdk-amd64/jre/lib/jfr.jar:/usr/lib/jvm/java-8-openjdk-amd64/jre/classes
[info] * -scalajs
[success] Total time: 0 s, completed 14 janv. 2020 23:57:46
```

File: sbt-dotty/sbt-test/sbt-dotty/i7897/test [hunk 4]
```diff
@@ -1,3 +1,4 @@
 > run
-$ copy-file build-no-fork.sbt build.sbt
+$ copy-file changes/build-no-fork.sbt build.sbt
```
Comment (nicolasstucki) [inline sbt-dotty/sbt-test/sbt-dotty/i7897/test]: This commit deleted the `build-no-fork.sbt` instead of moving it.

**Pull request merged**